addon/utils/query-cache.js

```diff
@@ -0,0 +1,69 @@
+
+const CACHE_DECAY_TIMEOUT = 5000;
+
+export default class QueryCache {
+  constructor() {
+    this._internalCache = Object.create(null);
+    this.decayId = null;
+  }
+
+  add(type, key, value) {
+    if (!key || !value) {
```

Comment:
It is unexpected that a false-like key or false-like value will silently do nothing.

My recommendation:

* if the key is invalid, throw
* allow caching a false-like value. For example, what if the cached value is supposed to be `null` or `false` or `undefined` (which seem perfectly valid).